api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -3,29 +3,24 @@
 import com.codeforcommunity.api.INotesProcessor;
 import com.codeforcommunity.api.IAuthProcessor;
 import com.codeforcommunity.auth.JWTAuthorizer;
-import com.codeforcommunity.dto.*;
 
-import com.codeforcommunity.exceptions.MissingRequestBodyException;
-import com.codeforcommunity.exceptions.RequestBodyMappingException;
 import com.codeforcommunity.rest.subrouter.AuthRouter;
 import com.codeforcommunity.rest.subrouter.CommonRouter;
+import com.codeforcommunity.rest.subrouter.FailureHandler;
 import com.codeforcommunity.rest.subrouter.NotesRouter;
 import io.vertx.core.Vertx;
 
 import io.vertx.core.http.HttpServerResponse;
-import io.vertx.core.json.JsonObject;
 import io.vertx.ext.web.Router;
-import io.vertx.ext.web.RoutingContext;
 
-import java.util.Optional;
 
 public class ApiRouter implements IRouter {
     private final CommonRouter commonRouter;
     private final NotesRouter notesRouter;
     private final AuthRouter authRouter;
 
-    public ApiRouter(INotesProcessor notesProcessor, IAuthProcessor authProcessor, JWTAuthorizer jwtAuthorizer) {
-        this.commonRouter = new CommonRouter(jwtAuthorizer);
+    public ApiRouter(INotesProcessor notesProcessor, IAuthProcessor authProcessor, JWTAuthorizer jwtAuthorizer, FailureHandler failureHandler) {
```

Comment:
Considering FailureHandler doesn't need anything special I would probably just construct in the CommonHandler where it's being used

Reply:
sounds good